api/gas/gas.go

```diff
@@ -22,4 +28,7 @@ type Estimator interface {
 	// required to get a transaction into one of the next few blocks (because
 	// blocks happen a lot faster).
 	EstimateGasPrice(context.Context) (pack.U256, error)
+
+	// EstimateGasLimit ...
+	EstimateGasLimit(TxType) (pack.U256, error)
```

Comment:
For now, we should keep estimating the gas limit as the responsibility of the user of the multichain. Until we have multiple use-cases in production, I do not think we should attempt to pre-empty what this interface will need to look like.